tutorial/nixify/1_nix_flake.md

```diff
@@ -0,0 +1,364 @@
+# Nixify your haskell project
+
+The purpose of this blog posts series is to simplify Haskell development workflow for you. We will begin by addressing the difficulties of working on a Haskell project without Nix, and then introduce [Nix flakes](https://nixos.wiki/wiki/Flakes) as the solution. If you're unfamiliar with Nix, we have [a quick introduction](https://haskell.flake.page/nix-rapid) available to help you get started quickly or you can take your time and explore it at [Zero to Nix](https://zero-to-nix.com). A basic understanding of the Nix expression language is assumed. Throughout the series, we will utilize a simple Haskell app called [todo-app](https://github.com/juspay/todo-app/tree/903c769d4bda0a8028fe3775415e9bdf29d80555) to illustrate how to build a Haskell project and automatically manage runtime dependencies such as databases (ie., [postgres](https://www.postgresql.org)) and and other services (here, we use [postgREST](https://postgrest.org/en/stable)), eliminating the need for any manual onboarding setup. This will allow us to highlight the advantages of using Nix.
+
+## Why Nixify?
+
+Why use Nix to develop a Haskell project rather than something like Stack or GHCup?
+
+- **Instant onboarding**: Projects have READMEs that describe how to setup the development environment but these instructions do not work the same way for 
+every developer and usually takes hours or days to setup. With Nix the setup is instantaneous and reproducible,[^nix] which means any new developer
+can get the development environment up and running with one command.
+- **Enhanced productivity**: More time spent on writing Haskell as Nix gives a fully working development environment with `nix develop`.
+- **Multi-platform**: Same configuration generally works on macOS,[^nm] Linux and WSL.
+
+[^nix]: Considering the packages are available in Nix for the host platform.
+[^nm]: Although macOS doesn't have first-class support in nixpkgs, [it is getting there](https://github.com/NixOS/nixpkgs/issues/116341).
+
+The rest of this blog post will provide a step-by-step demonstration of how to Nixify the todo-app project.
+
+
+## Introduce Flake
+
+To begin, clone the [todo-app](https://github.com/juspay/todo-app/tree/903c769d4bda0a8028fe3775415e9bdf29d80555) repository and checkout the specified commit.
+
+```sh
+git clone https://github.com/juspay/todo-app.git
+cd todo-app
+git checkout 903c769d4bda0a8028fe3775415e9bdf29d80555
+```
+
+Next, in the project's root directory, create a file named `flake.nix`. We will begin by setting up a basic template for the flake, which includes:
```

Comment:
it will be helpful to mention adding flake.nix to git staging with `git add flake.nix`, otherwise the error `error: source tree referenced by 'git+file:///Users/hao/Sandbox/todo-app?ref=HEAD&rev=903c769d4bda0a8028fe3775415e9bdf29d80555' does not contain a '/flake.nix' file` will be confusing to newcomer

Reply:
It is there in the asciinema video walkthrough. Having it explicitly mentioned will be better. Thanks!